File: frontend/src/api/artifacts.ts
/**
 * Typed API client for artifact download endpoints.
 * Returns presigned MinIO URLs for GDS, DEF, timing reports, and layout PNG.
 */
import { apiClient } from "./client";

// ---------------------------------------------------------------------------
// Response types (matching backend ArtifactURLs schema)
// ---------------------------------------------------------------------------

export interface ArtifactURLs {
  gds_url: string | null;
  def_url: string | null;
  timing_report_url: string | null;
  layout_png_url: string | null;
  run_id: string;
  expires_in_seconds: number;
}

// ---------------------------------------------------------------------------
// Artifact endpoints
// ---------------------------------------------------------------------------

/**
 * Get presigned download URLs for a completed run's artifacts.
 * Returns 404 if artifacts are not yet available (job still running or failed).
 */
export async function getArtifacts(runId: string): Promise<ArtifactURLs> {
  const { data } = await apiClient.get<ArtifactURLs>(`/jobs/${runId}/artifacts`);
  return data;
}

File: backend/app/schemas/artifacts.py
"""Pydantic schemas for artifact download endpoints."""
from pydantic import BaseModel


class ArtifactURLs(BaseModel):
    """Presigned download URLs for a completed run's artifacts.

    All URLs expire after expires_in_seconds (default 1 hour).
    Fields are None if the artifact does not exist in storage.
    """

    gds_url: str | None = None
    def_url: str | None = None
    timing_report_url: str | None = None
    layout_png_url: str | None = None
    run_id: str
    expires_in_seconds: int = 3600
